Fix openssl 3.x compatibility

FIPS_mode_set was merely used to clean up resources. This isn't necessary
anymore on the newer openssl version.

There's still a lot of functions that are used that were deprecated. I hope
upstream can fix those, as I'm unsure what to do with them.

--- tdesktop-3.1.0-full.orig/Telegram/SourceFiles/core/utils.cpp
+++ tdesktop-3.1.0-full/Telegram/SourceFiles/core/utils.cpp
@@ -124,7 +124,7 @@
 
 	void finish() {
 		CRYPTO_cleanup_all_ex_data();
-#ifndef LIBRESSL_VERSION_NUMBER
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER < 0x20000000L
 		FIPS_mode_set(0);
 #endif
 		ENGINE_cleanup();
